conwayste/src/client.rs

```diff
@@ -628,13 +628,41 @@ impl EventHandler for MainState {
 
         match current_screen {
             Screen::Intro => {
-                self.draw_intro(ctx)?;
+                let result = self.draw_intro(ctx);
+                let mut out_result: GameResult<()> = Ok(());
+                let mut do_return = false;
+                handle_error!{result,
+                    GameError => |e| {
+                        out_result = Err(e);
+                    },
+                    else => |e| {
+                        error!("Error from draw_intro: {}", e);
+                        do_return = true;
+                    }
+                };
+                if do_return {
+                    return out_result;
+                }
```

Comment:
I'm not seeing the point in having either `out_result` or `do_return`.

1) `out_result` is assigned to in the `GameError`, but the assigned value is not used afterwards.

2) When `do_return` is true, the `out_result` will always be `Ok(())`, so why have the variable at all?

Same goes for lines 651 to 665  in the case for `Screen::Run`.

Reply:
How about this. OK now?

Reply:
This makes much more sense about the intended purpose. Thanks. 

The only thing I see now is that when `do_return` is `true`, we bail out of the draw routine early, specifically skipping the call to `graphics::present()`. If the error persists between event loops, then this will result in the game appearing to have locked-up because we're aborting the frame updates. Doesn't seem like what we would want to happen. I don't think we should return early; printing the error should be enough for debugging; deciding what we do when we get the error is a separate discussion.

Reply:
Maybe it would be better to not return for `draw_universe` errors. I updated this so that `do_return` is removed.